packages/vue-apollo-composable/src/useMutation.ts

```diff
@@ -88,18 +92,28 @@ export function useMutation<
       currentOptions = options
     }
     const client = resolveClient(currentOptions.clientId)
+    if (!client) {
+      throw new Error('Apollo client instance not provided');
+    }
     error.value = null
     loading.value = true
     called.value = true
     try {
-      const result = await client.mutate<TResult>({
+      const mutationVariables: TVariables | undefined = [currentOptions.variables || variables]
```

Comment:
What was the reason for this change?

Reply:
The original code produced the following error in strict mode

```ts
variables: {
  ...currentOptions.variables || {},
  ...variables || {},
}
```

```
[tsserver 2322] [E] Type 'NonNullable<TVariables> | (NonNullable<TVariables> & NonNullable<TVariables>) | (NonNullable<TVariables> & {}) | {}' is not assignable to type 'TVariables | undefined'.
  Type '{}' is not assignable to type 'TVariables'.
    '{}' is assignable to the constraint of type 'TVariables', but 'TVariables' could be instantiated with a different subtype of constraint 'Record<string, any>'.
```

Typescript was telling me it wanted only `TVariables | undefined` and not `{}`. So I collect all possible variable declarations in an array `Array<TVariables | undefined>`, filter those that are defined `Array<TVariables>` and then merge them using `reduce`.

Reply:
I found a mistake on my part. Previously, I was constructing this array

```
[currentOptions.variables || variables]
```

However I really meant this

```
[currentOptions.variables, variables]
```

It's fixed now

Reply:
I don't really get why we need all of this logic just to merge two objects. I don't want to make the code unreadable just to make typescript happy... :confused:

Reply:
There are alternatives. The simplest one is not to merge them, `variables: variables ?? currentOptions.variables`.

Another alternative involves type casting. We lose a little bit of type safety in exchange of briefness.

```ts
variables: (variables ?? currentOptions.variables)
  ? {
    ...(currentOptions.variables as NonNullable<TVariables>),
    ...(variables as NonNullable<TVariables>),
  }
  : undefined,
```

Reply:
I think you will like the type casting idea, so I already applied it in the last rebase.